### PR TITLE
Add `sasl_SUITE` and `sasl_external_SUITE` to dynamic domains testing 

### PR DESCRIPTION
**Pull request by Premwoik**
Enable sasl_SUITE and sasl_external_SUITE for dynamic domains.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/72569) / small_tests / 64689f0f1685485e7784dae091ea2297a061b8c0
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3269/72569/small_tests.24.0.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3269/72569/small_tests.24.0.4/small/ct_run.mongooseim@localhost.2021-09-14_11.58.40/lib.mongooseim.logs/run.2021-09-14_11.58.40/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/72817) / small_tests / 1173945a3970594c7d412050c607f8ccef1f902d
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3269/72817/small_tests.24.0.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3269/72817/small_tests.24.0.4/small/ct_run.mongooseim@localhost.2021-09-14_19.43.29/lib.mongooseim.logs/run.2021-09-14_19.43.29/suite.log.html)

**Review by chrzaszcz — CHANGES_REQUESTED** [review]
I have no comments regarding `sasl_SUITE`, but `sasl_external_SUITE` seems to use hardcoded `localhost` as the domain name in many places. This needs to be changed to the correct domain. (`domain() -> ct:get_config({hosts, mim, domain})`). You can look at the published test report from CircleCI and see that `localhost` is used: https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3269/72832/pgsql_mnesia.24.0-1/big/ct_run.test@default-bcacc94f-cf46-4125-9779-f4dfd35285dd.2021-09-14_20.02.03/big_tests.tests.sasl_external_SUITE.logs/run.2021-09-14_20.09.40/sasl_external_suite.self_signed_cert_fails_to_authenticate_with_tls.18259.html

File: big_tests/tests/sasl_external_SUITE.erl [hunk 1]
```diff
@@ -469,3 +471,15 @@ replace_addrs(undefined) ->
...
+domain() ->
+    ct:get_config({hosts, mim, domain}).
+
+-spec add_domain_str(User :: string()) -> string().
+add_domain_str(User) ->
+    User ++ "@" ++ binary:bin_to_list(domain()).
+
+-spec add_domain(User :: binary()) -> binary().
+add_domain(User) ->
+    <<User/binary, <<"@">>/binary, (domain())/binary>>.
```
Comment (vkatsuba) [inline big_tests/tests/sasl_external_SUITE.erl:485]:
```suggestion
    <<User/binary, "@", (domain())/binary>>.
```

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/73126) / small_tests / 8aa78bd362dafbfe9d7066bb05106bf4916e52fa
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3269/73126/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3269/73126/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-09-16_10.17.43/lib.mongooseim.logs/run.2021-09-16_10.17.44/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/73151) / small_tests / 502c805fd083ca4f49e3f17c5b2f4487f3bcf356
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3269/73151/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3269/73151/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-09-16_11.10.16/lib.mongooseim.logs/run.2021-09-16_11.10.16/suite.log.html)

File: big_tests/tests/sasl_external_SUITE.erl [hunk 2]
```diff
@@ -420,7 +421,8 @@ generate_user(C, User, Transport) ->
     UserCert = maps:get(User, Certs),
 
     Common = [{username, list_to_binary(User)},
-	      {server, <<"localhost">>},
+	      {server, domain()},
```
Comment (chrzaszcz) [inline big_tests/tests/sasl_external_SUITE.erl:424]: Could you realign these lines to make all the opening `{` have the same indentation as `{username` ?

File: big_tests/tests/sasl_external_SUITE.erl [hunk 3]
```diff
@@ -469,3 +471,15 @@ replace_addrs(undefined) ->
 replace_addrs(Addresses) ->
     lists:map( fun(Addr) -> [User, Hostname] = binary:split(list_to_binary(Addr), <<"@">>),
                             binary_to_list(<<User/binary, <<"-self-signed@">>/binary, Hostname/binary>>) end, Addresses).
+
+-spec domain() -> mongooseim:domain_name().
```
Comment (chrzaszcz) [inline big_tests/tests/sasl_external_SUITE.erl:475]: I think the returned type is not available in big tests (it is undefined).
```suggestion
-spec domain() -> jid:lserver().
```

**Review by chrzaszcz — COMMENTED** [review]
The code looks good in general. I have two comments.

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/73237) / small_tests / 51f59bfb791d258cd1b8eb68dfe74fa3485114f4
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3269/73237/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3269/73237/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-09-16_12.23.36/lib.mongooseim.logs/run.2021-09-16_12.23.36/suite.log.html)

**Review by chrzaszcz — APPROVED** [review]
Looks good!

**Pull request merged**